File: 1752. Check if Array Is Sorted and Rotated.java
import java.util.*;
class Solution {
    public boolean check(int[] nums) {
        int n=nums.length;
        if(n<1 || n>100)
        System.exit(0);
        for(int i=0;i<n;i++)
        {
            if(nums[i]<1 || nums[i]>100)
            System.exit(0);
        }
        int pivot=0;
        for(int i=0;i<n-1;i++)
        {
            if(nums[i]>nums[i+1])
            {
                pivot=i+1;
                break;
            }
        }
        if(pivot==0)
        return true;
        else 
        {
            for(int i=0;i<n-1;i++)
            {
                if(nums[(i+pivot)%n]>nums[(i+pivot+1)%n])
                return false;
            }
            return true;
        }
    }
}

File: 1752. Check if Array Is Sorted and Rotated.rs
impl Solution {
    pub fn check(nums: Vec<i32>) -> bool {
        let mut c = 0;
        for i in 1..nums.len() {
            if nums[i-1] > nums[i] {
                c+=1;
            }
            if c > 1 {
                return false
            }
        }
        if nums[0] < nums[nums.len()-1] {
            c+=1;
        }
        c<=1
    }
}
